pkg/splunk/splunk.go

```diff
@@ -0,0 +1,218 @@
+		splunkDsn:             dsn,
+		splunkIndex:           index,
+		correlationID:         correlationID,
+		postMessagesBatchSize: 1000,
+		sendLogs:              sendLogs,
+	}
+	return nil
+}
+
+func Send(customTelemetryData *telemetry.CustomData, logCollector *log.CollectorHook) error {
+	// Sends telemetry and or additionally logging data to Splunk
+	telemetryData := prepareTelemetry(*customTelemetryData)
+	messagesLen := len(logCollector.Messages)
+	// TODO: Logic for errorCategory (undefined, service, infrastructure)
+	if telemetryData.ErrorCode == "0" {
+		// Successful run, we only send the telemetry data, no logging information
+		err := tryPostMessages(telemetryData, []log.Message{})
+		if err != nil {
+			log.Entry().WithError(err).WithField("module", "logger/splunk").Warn("Error while sending logs")
+		}
+		return nil
+	} else if telemetryData.ErrorCode == "1" && !SplunkClient.sendLogs {
+		// Failure run and we do not want to send the logs
+		err := tryPostMessages(telemetryData, []log.Message{})
+		if err != nil {
+			log.Entry().WithError(err).WithField("module", "logger/splunk").Warn("Error while sending logs")
+		}
+	} else {
+		// ErrorCode indicates an error in the step, so we want to send all the logs with telemetry
+		for i := 0; i < messagesLen; i += SplunkClient.postMessagesBatchSize {
+			upperBound := i + SplunkClient.postMessagesBatchSize
+			if upperBound > messagesLen {
+				upperBound = messagesLen
+			}
+			err := tryPostMessages(telemetryData, logCollector.Messages[i:upperBound])
+			if err != nil {
+				log.Entry().WithError(err).WithField("module", "logger/splunk").Warn("Error while sending logs")
+			}
+		}
+	}
+	return nil
+}
+
+func readPipelineEnvironment() (string, string, string, string) {
+
+	// TODO: Dependent on a groovy step, which creates the folder.
+	// TODO: go git step here to read infos from git folder later add this to telemetry package?
+	contentCommitId, err := ioutil.ReadFile(".pipeline/commonPipelineEnvironment/git/commitId")
+	if err != nil {
+		log.Entry().Warnf("Could not read commitId file. %v", err)
+		contentCommitId = []byte("N/A")
+	}
+	commitHash := string(contentCommitId)
+
+	contentBranch, err := ioutil.ReadFile(".pipeline/commonPipelineEnvironment/git/branch")
+	if err != nil {
+		log.Entry().Warnf("Could not read branch file. %v", err)
+		contentBranch = []byte("N/A")
+	}
+	branch := string(contentBranch)
+
+	contentRepository, err := ioutil.ReadFile(".pipeline/commonPipelineEnvironment/github/repository")
+	if err != nil {
+		log.Entry().Warnf("Could not read branch file. %v", err)
+		contentRepository = []byte("N/A")
+	}
+	gitRepository := string(contentRepository)
+
+	contentGitHubOwner, err := ioutil.ReadFile(".pipeline/commonPipelineEnvironment/github/owner")
+	if err != nil {
+		log.Entry().Warnf("Could not read branch file. %v", err)
+		contentGitHubOwner = []byte("N/A")
+	}
+	gitOwner := string(contentGitHubOwner)
+
+	return commitHash, branch, gitOwner, gitRepository
```

Comment:
can be refactored to something like this:

```suggestion
	return readCPE('git/commitId'), readCPE('git/commitId'), readCPE('git/commitId'), readCPE('git/commitId')
```